lib/acl-checker.js

```diff
@@ -0,0 +1,230 @@
+'use strict'
+
+const async = require('async')
+const path = require('path')
+const url = require('url')
+
+const DEFAULT_ACL_SUFFIX = '.acl'
+
+class ACLChecker {
+  constructor (options = {}) {
+    this.debug = options.debug || console.log
+    this.fetch = options.fetch
+    this.strictOrigin = options.strictOrigin
+    this.suffix = options.suffix || DEFAULT_ACL_SUFFIX
+  }
```

Comment:
can we still keep an optional match?
in principle, I wanted this to be able to work with any match function, in fact, this piece of code was working with rdflib and rdf-ext